modules/video/src/ecc.cpp

```diff
@@ -309,16 +309,53 @@ static void update_warping_matrix_ECC (Mat& map_matrix, const Mat& update, const
 }
 
 
+/** Function that computes enhanced corelation coefficient from Georgios et.al. 2008
+*   See https://github.com/opencv/opencv/issues/12432
+*/
+double cv::computeECC(InputArray templateImage, InputArray inputImage, InputArray inputMask)
+{
+    Mat src = templateImage.getMat();
+    Mat dst = inputImage.getMat();
```

Comment:
> dst = inputImage

This doesn't look consistent.

It is better to declare parameters with underscore `_` suffix (in .cpp file only, keep .hpp without this hack). For "Mat" use original names.

```
double cv::computeECC(InputArray templateImage_, InputArray inputImage_, InputArray inputMask)
{
    Mat templateImage = templateImage_.getMat();
    Mat inputImage = inputImage_.getMat();
```

Anyway, you don't need to extract `Mat` from `InputArray`. Just use "InputArray" directly:
- `.cols` -> `.cols()`
- `.rows` -> `.rows()`
- or `Mat::ones(src.rows, src.cols, ...)` -> `Mat::ones(templateImage.size(), ...)`

Reply:
Thanks for your patient guidance. Done.